pages/api/performances/edit.js

```diff
@@ -0,0 +1,65 @@
+import prisma from '@prisma/index'; // Prisma client
+import { getSession } from 'next-auth/client'; // Session handling
+
+export default async (req, res) => {
+  // Collect session from request
+  const session = await getSession({ req });
+
+  // If not authenticated or is not an admin
+  if (!session) {
+    return res.status(401).end();
+  }
+
+  // Collect performance information from request body
+  const {
+    id,
+    name,
+    academicLevel,
+    performers,
+    choreographers,
+    competitionLevel,
+    danceSize,
+    danceEntry,
+    danceStyle,
+    danceTitle,
+    eventID,
+    schoolID,
+  } = req.body;
+
+  if (!id) {
+    return res.status(400).json({
+      error: 'Required performance id not provided',
+    });
+  }
+  console.log(danceTitle);
+
+  const editData = {};
+  if (name) editData.name = name;
```

Comment:
Is there a better way of doing this LOL

Reply:
for edit my impression was the frontend passes in all the data every time they call edit.  no need to support per field 'patching' I think.